src/cpp/session/SessionModuleContext.cpp

```diff
@@ -95,13 +95,65 @@
 
 #include "session-config.h"
 
+#ifdef _WIN32
+# define kPathSeparator ';'
+#else
+# define kPathSeparator ':'
+#endif
+
 using namespace rstudio::core;
 using namespace boost::placeholders;
 
 namespace rstudio {
 namespace session {   
 namespace module_context {
 
+namespace {
+
+Error modifyPath(const core::FilePath& path, bool append)
+{
+   Error error;
+
+   std::string currentPath;
+   error = r::exec::RFunction("base:::Sys.getenv")
+         .addParam("PATH")
+         .call(&currentPath);
+
+   if (error)
+      return error;
+
+   std::string pathString = string_utils::utf8ToSystem(path.getAbsolutePath());
+
+#ifdef _WIN32
+   std::replace(pathString.begin(), pathString.end(), '/', '\\');
```

Comment:
Is there a particular reason for changing this now in Windows? We've always had the forward slashes in the path before on Windows, and now with this we sometimes get a mix of forward and backslashes, for example:

```
 "C:\\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\\rtools42/usr/bin;C:\\R\\R-4.2.1\\bin\\x64;
```
just wondering if you suddenly encountered some issue with the forward slashes that prompted this change because I think most of our codebase tends to assume that the forward slashes will work the same on Windows and we specifically don't need to convert to native path separators

Reply:
There isn't a strong reason, but _usually_ things on the Windows PATH should use backslashes. In practice, Windows allows both separators, so it's more hygiene than anything.

> because I think most of our codebase tends to assume that the forward slashes will work the same on Windows and we specifically don't need to convert to native path separators

That's true, but the PATH is a somewhat special case where IMHO the "correct" thing to do is to use backslashes. However, whenever we want to handle paths internally (or return those paths to the client) we should definitely use forward slashes.

Reply:
I was curious where the mixed slashes for rtools42 were coming from, and that's from R itself:

https://github.com/wch/r-source/blob/4ca6439c1ffc76958592455c44d83f95d5854b2a/src/library/profile/Rprofile.windows#L64-L76

I would argue that those should be backslashes instead, but it's not a huge deal ultimately.